src/map/battle.cpp

```diff
@@ -2373,21 +2373,27 @@ static bool is_attack_left_handed(struct block_list *src, int skill_id)
  */
 static bool is_attack_critical(struct Damage* wd, struct block_list *src, struct block_list *target, int skill_id, int skill_lv, bool first_call)
 {
-	struct status_data *sstatus = status_get_status_data(src);
-	struct status_data *tstatus = status_get_status_data(target);
-	struct status_change *sc = status_get_sc(src);
-	struct status_change *tsc = status_get_sc(target);
-	struct map_session_data *sd = BL_CAST(BL_PC, src);
-	struct map_session_data *tsd = BL_CAST(BL_PC, target);
-
 	if (!first_call)
 		return (wd->type == DMG_CRITICAL || wd->type == DMG_MULTI_HIT_CRITICAL);
 
-	if (skill_id == NPC_CRITICALSLASH || skill_id == LG_PINPOINTATTACK) //Always critical skills
-		return true;
+	if( skill_id && !skill_get_nk(skill_id)&NK_CRITICAL )
+		return false;
 
-	if( sstatus->cri && ( !skill_id || skill_get_nk(skill_id)&NK_CRITICAL ) )
+	switch(skill_id) { //Always critical skills
```

Comment:
unnecessary, moved to skill_db already

Reply:
but if we removed them from here , they will enter the `sstatus->cri` check for a chance to crit
while they should always crit without checking for the chance of crit , or that what i understood from the code

in https://irowiki.org/wiki/Pinpoint_Attack
`Always deals Critical Damage, regardless of user's Critical Hit Rate.`

and https://irowiki.org/wiki/NPC_CRITICALSLASH
`Deals a Critical Attack at a target.`

Reply:
revert the skill db then

Reply:
the skill db idea , is very useful , if someone want to disable the skill crit , he can just edit the skill_db
the only problem in the skill db i think is we can't control the crit chance from there

i think it's better to leave the skill_can_crit option in the skill_db , while here we control the crit chance.
ofc this only my opinion xD

Reply:
I share @sader1992's opinion on this. Having it in skill db is fine, but those 100% crit rate checks are required.